marathon/src/main/scala/io/buoyant/marathon/v2/AppIdNamer.scala

```diff
@@ -42,8 +42,10 @@ class AppIdNamer(
       val possibleIds = (1 to path.size).map(path.take(_))
       appsActivity.map { apps =>
         Trace.recordBinary("marathon.path", path.show)
-        val found = possibleIds.collectFirst {
-          case app if apps(app) =>
+        val found = possibleIds.toStream.map { app =>
```

Comment:
Another option would be to call `.toLowerCase` before `Path.read` in `Api.toAppIds` to make sure all ids in `appsActivity` are lowercase.  Then we can convert `path` to lowercase by doing 

```
val Path.Utf8(segs@_*) = path
val lowercasePath = Path.Utf8(segs.map(_.toLowerCase): _*)
```

this way we can avoid relying on `path.show` which is probably unreliable for equality